tests/test_diagnostics.py

```diff
@@ -0,0 +1,58 @@
+"""Test MCMC diagnoistics."""
+import jax
+import numpy as np
+import pytest
+
+import blackjax.diagnostics as diagnostics
+
+test_cases = [
+    {
+        "chain_axis": 0,
+        "sample_axis": 1,
+    },
+    {
+        "chain_axis": 1,
+        "sample_axis": 0,
+    },
+    {
+        "chain_axis": 0,
+        "sample_axis": -1,
+    },
+    {
+        "chain_axis": -1,
+        "sample_axis": 0,
+    },
+]
+
+
+def insert_list(input_list, loc, elem):
+    if loc == -1:
+        input_list.append(elem)
+    else:
+        input_list.insert(loc, elem)
+    return input_list
+
+
+@pytest.mark.parametrize("case", test_cases)
+@pytest.mark.parametrize("nchain", [2, 10])
+@pytest.mark.parametrize("event_shape", [(), (3,), (5, 7)])
+def test_rhat_ess(case, nchain, event_shape):
+    rng_key = jax.random.PRNGKey(32)
+    n_samples = 5000
```

Comment:
nitpicking: `n_samples` and `nchain` use different naming conventions

Reply:
I prefer `n_chains` (or better `num_chains`)

Reply:
Done